### PR TITLE
GitHub template for rollout issues

### LINKED ISSUE #2735: Create issue template for arcade-services rollouts
The process of weekly rollouts of our repos is well defined and described on the [Rollout](https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/831/Rollout) wiki page. With the transition of arcade-services` repo to our team, we've also become responsible for the repo production rollouts.

As part of the rollout ownership transition, we've agreed (as the team) to introduce some changes to the process, primarily to adjust the process to our working hours.

**The goal** of this issue is to create an issue template for the rollout process in this repo that will be used to instantiate a tracking issue for every rollout.

There are also some other improvements proposed to the rollout process, described at #2660.

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [x] Internal Infrastructure Improvements
### Release Note Description
Internal improvement of the release process management

### PR DESCRIPTION
**Pull request by tkapin**
[Resolves #2735](https://github.com/dotnet/arcade-services/issues/2735)

There are actually two ways how to implement the issue templates on GitHub - 1) simple markdown templates and 2) YAML-based issue forms. I've investigated both and the more structured YAML forms seem to be an overkill for our purpose that would just be more difficult to read, hence I'm going with the simple Markdown based template.

@premun - I've unfortunately lost the original notes I've made so some of the steps (especially for the rollback) that we've talked about might be missing

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+name: Rollout issue
+about: 'Create issue for arcade-services rollout'
+title: Rollout DD-MM-YYYY
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:4]:
```suggestion
title: Rollout YYYY-MM-DD
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+---
+name: Rollout issue
+about: 'Create issue for arcade-services rollout'
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:3]:
```suggestion
about: Create issue for arcade-services rollout
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+title: Rollout DD-MM-YYYY
+labels: Rollout
+assignees: ''
+
+---
+
+# Purpose
+
+This issue tracks the `arcade-services` repository rollout. On top of the [Rollout](https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/831/Rollout) instructions described on the wiki, it provides the person responsible for the rollout checklist of the steps that should to be performed to rollout services in this repository. All relevant information, including the rollout PR, issues encountered during the rollout and steps taken to resolve them should be linked or added to this issue to keep full audit trail of changes rolled out to production.
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:12]:
```suggestion
This issue tracks the `arcade-services` repository rollout. On top of the [Rollout](https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/831/Rollout) instructions described on the wiki, it provides the person responsible for the rollout checklist of the steps that need to be performed to rollout services in this repository. All relevant information, including the rollout PR, issues encountered during the rollout and steps taken to resolve them should be linked or added to this issue to keep full audit trail of changes rolled out to production.
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+# Process
+
+## Build status check (Monday)
+- [ ] Check the status of the [dotnet-arcade-services-weekly](https://dev.azure.com/dnceng/internal/_build?definitionId=993) pipeline
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:17]:
```suggestion
- [ ] Check the status of the [dotnet-arcade-services-weekly](https://dev.azure.com/dnceng/internal/_build?definitionId=993) pipeline. Rotate any secrets that need manual rotation.
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Ensure the [arcade-services-internal-ci](https://dev.azure.com/dnceng/internal/_build?definitionId=252) pipeline is green
+- [ ] In case there is a problem with the CI build, notify the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:19]: I think this should happen on Tuesday. On Monday we should still have a chance to fix the build.

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+## Rollout preparation (Tuesday)
+- [ ] Wait for the vendor to prepare the rollout. This includes a thread on the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47), rollout issue in [AzDO](https://dev.azure.com/dnceng/internal/_workitems/) and the rollout PR
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:22]:
```suggestion
- [ ] Wait for vendors to prepare the rollout: 
    - Thread on the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
    - Rollout issue in [AzDO](https://dev.azure.com/dnceng/internal/_workitems/)
    - Rollout PR in `arcade-services`
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Link the rollout PR to the "Rollout data" section of this issue
+- [ ] Double-check that the release notes contain all information
+- [ ] Merge the already prepard rollout PR
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:25]:
```suggestion
- [ ] Merge the already prepared rollout PR
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Ensure the build is green and stops on the "Approval" phase
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:26]:
```suggestion
- [ ] Ensure the build is green and stops at the "Approval" phase
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+## Rollout day (Wednesday)
+- [ ] Approve the rollout build (that has been already started the day before)
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:29]:
```suggestion
- [ ] Approve the `Approval` stage of the rollout build (that has been already started the day before)
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Monitor the rollout build for failures. The [Maestro exceptions](https://ms.portal.azure.com/#view/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fmaestro-prod-cluster%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fmaestro-prod/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAAz2MOw6DMBBE%252B5xiSlsiRZDS5i7GjGQXu0brRSSIwyekoH4fvjMXr0377cBWaIRXYfckC17QtoV4H%252Bcf7KtIsroTua3qIWL6YKoaLn%252FA4ylxgNBLOxOjzrT%252FMJdk%252FgV08ryabQAAAA%253D%253D) query might help in diagnosing issues.
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:30]:
```suggestion
- [ ] Monitor the rollout build for failures.
    > Note: The [Maestro exceptions query](https://ms.portal.azure.com/#view/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fmaestro-prod-cluster%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fmaestro-prod/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAAz2MOw6DMBBE%252B5xiSlsiRZDS5i7GjGQXu0brRSSIwyekoH4fvjMXr0377cBWaIRXYfckC17QtoV4H%252Bcf7KtIsroTua3qIWL6YKoaLn%252FA4ylxgNBLOxOjzrT%252FMJdk%252FgV08ryabQAAAA%253D%253D) might help in diagnosing issues.
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Keep track of any issues encountered during the rollout either directly in this issue, or in a dedicated issue linked to this issue
+- [ ] Update the rollout stats in the "Stats" section below
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:32]:
```suggestion
- [ ] Update the rollout stats in the [Stats](#stats) section below
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+# Process
+
+## Build status check (Monday)
+- [ ] Check the status of the [dotnet-arcade-services-weekly](https://dev.azure.com/dnceng/internal/_build?definitionId=993) pipeline
+- [ ] Ensure the [arcade-services-internal-ci](https://dev.azure.com/dnceng/internal/_build?definitionId=252) pipeline is green
+- [ ] In case there is a problem with the CI build, notify the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
+
+## Rollout preparation (Tuesday)
+- [ ] Wait for the vendor to prepare the rollout. This includes a thread on the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47), rollout issue in [AzDO](https://dev.azure.com/dnceng/internal/_workitems/) and the rollout PR
+- [ ] Link the rollout PR to the "Rollout data" section of this issue
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:23]:
```suggestion
- [ ] Link the rollout PR to the [Rollout data](#rollout-data) section of this issue
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Monitor the rollout build for failures. The [Maestro exceptions](https://ms.portal.azure.com/#view/Microsoft_OperationsManagementSuite_Workspace/Logs.ReactView/resourceId/%2Fsubscriptions%2F68672ab8-de0c-40f1-8d1b-ffb20bd62c0f%2FresourceGroups%2Fmaestro-prod-cluster%2Fproviders%2Fmicrosoft.insights%2Fcomponents%2Fmaestro-prod/source/LogsBlade.AnalyticsShareLinkToQuery/q/H4sIAAAAAAAAAz2MOw6DMBBE%252B5xiSlsiRZDS5i7GjGQXu0brRSSIwyekoH4fvjMXr0377cBWaIRXYfckC17QtoV4H%252Bcf7KtIsroTua3qIWL6YKoaLn%252FA4ylxgNBLOxOjzrT%252FMJdk%252FgV08ryabQAAAA%253D%253D) query might help in diagnosing issues.
+- [ ] Keep track of any issues encountered during the rollout either directly in this issue, or in a dedicated issue linked to this issue
+- [ ] Update the rollout stats in the "Stats" section below
+- [ ] Notify the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
+- [ ]  Close this issue with closing comment describing a high-level summary of issues encountered during the rollout
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:34]:
```suggestion
- [ ] Close this issue with closing comment describing a high-level summary of issues encountered during the rollout
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+- [ ] Ensure the [arcade-services-internal-ci](https://dev.azure.com/dnceng/internal/_build?definitionId=252) pipeline is green
+- [ ] In case there is a problem with the CI build, notify the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
+
+## Rollout preparation (Tuesday)
+- [ ] Wait for the vendor to prepare the rollout. This includes a thread on the [Rollout channel](https://teams.microsoft.com/l/channel/19%3a72e283b51f9e4567ba24a35328562df4%40thread.skype/Rollout?groupId=147df318-61de-4f04-8f7b-ecd328c256bb&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47), rollout issue in [AzDO](https://dev.azure.com/dnceng/internal/_workitems/) and the rollout PR
+- [ ] Link the rollout PR to the "Rollout data" section of this issue
+- [ ] Double-check that the release notes contain all information
+- [ ] Merge the already prepard rollout PR
+- [ ] Ensure the build is green and stops on the "Approval" phase
+
```
Comment (premun) [inline .github/ISSUE_TEMPLATE/rollout-issue.md:27]:
```suggestion
- [ ] Send an email to **.NET Core Engineering Partners** with the planned rollout date and release notes.

```

@tkapin do we want this?

**Review by premun — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/rollout-issue.md (hunk 3 above)
Comment (dkurepa) [reply .github/ISSUE_TEMPLATE/rollout-issue.md:12]: ads
```suggestion
This issue tracks the `arcade-services` repository rollout. On top of the [Rollout](https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/831/Rollout) instructions described on the wiki, it provides the person responsible for the rollout a checklist of the steps that need to be performed to rollout services in this repository. All relevant information, including the rollout PR, issues encountered during the rollout and steps taken to resolve them should be linked or added to this issue to keep a full audit trail of changes rolled out to production.
```

File: .github/ISSUE_TEMPLATE/rollout-issue.md (hunk 14 above)
Comment (premun) [reply .github/ISSUE_TEMPLATE/rollout-issue.md:27]: I see we reference the email later in the rollback section so I presume we do?

File: .github/ISSUE_TEMPLATE/rollout-issue.md (hunk 2 above)
Comment (tkapin) [reply .github/ISSUE_TEMPLATE/rollout-issue.md:3]: This was created by the GitHub templating engine itself, but we can give it a try and remove it to make it consistent with the other fields.

File: .github/ISSUE_TEMPLATE/rollout-issue.md (hunk 14 above)
Comment (tkapin) [reply .github/ISSUE_TEMPLATE/rollout-issue.md:27]: Let's sync on this tomorrow again. I believe we agreed on these emails not having much value, but I'd like to check with you again on this.

File: .github/ISSUE_TEMPLATE/rollout-issue.md (hunk 4 above)
Comment (tkapin) [reply .github/ISSUE_TEMPLATE/rollout-issue.md:17]: Added as separate check as it's another well-defined activity

**Review by premun — APPROVED** [review]

**Pull request merged**